core/common/src/main/java/alluxio/underfs/ObjectUnderFileSystem.java

```diff
@@ -544,6 +518,31 @@ protected abstract ObjectListingChunk getObjectListingChunk(String key, boolean
       throws IOException;
 
   /**
+   * Gets a (partial) object listing and attempt to create non-empty directory breadcrumb.
+   *
+   * @param path of pseudo-directory
+   * @param recursive whether to request immediate children only, or all descendants
+   * @return null if the prefix is not found or if an IO error occurs
+   */
+  protected ObjectListingChunk getObjectListingChunkAndCreateNonEmpty(String path,
+      boolean recursive) {
+    // Check if anything begins with <folder_path>/
+    String dir = stripPrefixIfPresent(path);
+    try {
+      ObjectListingChunk objs = getObjectListingChunk(dir, recursive);
+      // If there are, this is a folder and we can create the necessary metadata
+      if (objs != null && objs.getObjectNames() != null && objs.getObjectNames().length > 0) {
+        // If the breadcrumb exists, this is a no-op
+        mkdirsInternal(dir);
+        return objs;
+      }
+    } catch (IOException e) {
+      // Return null if an exception occurs
```

Comment:
thanks. exceptions are not expected, i improved the documentation to make this clear.